### ISSUE TITLE
Jupyter notebooks (v5.3.1) do not remain in "Running" state with notebook extensions installed

### ISSUE THREAD

**Issue by pybokeh** [jupyter/notebook#3245]
Hello, I apologize, I am not a developer type, so not sure what is all the necessary information I should be providing to help debug this issue.  I can duplicate this problem both on Windows 7 and Ubuntu 16.04 with python 3.6.2 virtual environment installed via pip/pipenv, not with Ananconda/miniconda.  I installed notebook extensions per their Github repo [instructions](https://github.com/ipython-contrib/jupyter_contrib_nbextensions) and installed/enabled extensions with the nb extensions configurator.

Essentially, when I upgraded my notebook from 5.2.1 to 5.3.1, my notebooks that I open up, do not stay "Running".  On Windows 7 at work, the top right of my notebook would say "kernel error".  On Ubuntu 16.04 machine at home, the top right would say "Forbidden".  But in both cases (or OS), my notebooks won't stay "Running" and if I then proceed to reload or close and re-open the same notebook, I am sent to the log-in screen.  My notebook would then launch, but then won't stay "Running" again.  I can view my "Running" tab, and see that indeed, no notebooks are "Running".  
  
I then downgraded to notebook version 5.2.1, and now I do not have the above problem.  I also did this:  I reinstalled 5.3.1 and then uninstalled my notebook extensions, the problem also went away.

So it seems that perhaps the current notebook extensions are not compatible with notebook version 5.3.1 or are causing problems (maybe I should open a Github issue at the notebook extensions repo??)

Below are 2 sets of output from my terminal from which I launched my notebook server.

**With notebook version 5.3.1 and notebook extensions installed:**

```[W 13:19:04.626 NotebookApp] server_extensions is deprecated, use nbserver_extensions
[I 13:19:05.009 NotebookApp] [jupyter_nbextensions_configurator] enabled 0.4.0
[I 13:19:05.011 NotebookApp] Serving notebooks from local directory: /home/pybokeh/Dropbox/python/jupyter_notebooks
[I 13:19:05.011 NotebookApp] 0 active kernels
[I 13:19:05.011 NotebookApp] The Jupyter Notebook is running at:
[I 13:19:05.011 NotebookApp] http://localhost:8888/jupyter/
[I 13:19:05.011 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 13:19:08.512 NotebookApp] Forbidden
[W 13:19:08.513 NotebookApp] 403 GET /jupyter/api/sessions?_=1516645026557 (127.0.0.1) 2.20ms referer=http://localhost:8888/jupyter/tree/dask
[W 13:19:08.518 NotebookApp] Forbidden
[W 13:19:08.519 NotebookApp] 403 GET /jupyter/api/terminals?_=1516645026558 (127.0.0.1) 1.41ms referer=http://localhost:8888/jupyter/tree/dask
[I 13:19:11.788 NotebookApp] 302 GET /jupyter/ (127.0.0.1) 0.65ms
[I 13:19:11.813 NotebookApp] 302 GET /jupyter/tree? (127.0.0.1) 1.02ms
[I 13:19:15.319 NotebookApp] 302 POST /jupyter/login?next=%2Fjupyter%2Ftree%3F (127.0.0.1) 2.44ms
[W 13:19:29.924 NotebookApp] Notebook dask/dask_dataframe_csv_example.ipynb is not trusted
[I 13:19:31.248 NotebookApp] Kernel started: 846f9b9d-9ac2-4aad-a199-7212ae0ff539
[W 13:19:31.359 NotebookApp] 404 GET /jupyter/nbextensions/dragdrop/main.js?v=20180122131904 (127.0.0.1) 17.36ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:19:31.690 NotebookApp] 404 GET /jupyter/nbextensions/search-replace/main.js?v=20180122131904 (127.0.0.1) 2.35ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:19:31.749 NotebookApp] 404 GET /jupyter/nbextensions/vim_binding/vim_binding.js?v=20180122131904 (127.0.0.1) 8.63ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:19:31.867 NotebookApp] 404 GET /nbextensions/snippets/snippets.json (127.0.0.1) 2.61ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:19:32.164 NotebookApp] 404 GET /jupyter/nbextensions/widgets/notebook/js/extension.js?v=20180122131904 (127.0.0.1) 2.12ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[I 13:19:32.273 NotebookApp] Adapting to protocol v5.1 for kernel 846f9b9d-9ac2-4aad-a199-7212ae0ff539
[W 13:19:44.497 NotebookApp] Forbidden
[W 13:19:44.498 NotebookApp] 403 GET /jupyter/api/sessions?_=1516645155607 (127.0.0.1) 1.44ms referer=http://localhost:8888/jupyter/tree/dask
[W 13:19:44.503 NotebookApp] Forbidden
[W 13:19:44.504 NotebookApp] 403 GET /jupyter/api/terminals?_=1516645155608 (127.0.0.1) 1.33ms referer=http://localhost:8888/jupyter/tree/dask
```

**With notebook version 5.2.1 installed and with extensions installed:**

```[W 13:21:27.248 NotebookApp] server_extensions is deprecated, use nbserver_extensions
[I 13:21:27.630 NotebookApp] [jupyter_nbextensions_configurator] enabled 0.4.0
[I 13:21:27.633 NotebookApp] Serving notebooks from local directory: /home/pybokeh/Dropbox/python/jupyter_notebooks
[I 13:21:27.633 NotebookApp] 0 active kernels
[I 13:21:27.633 NotebookApp] The Jupyter Notebook is running at:
[I 13:21:27.633 NotebookApp] http://localhost:8888/jupyter/
[I 13:21:27.633 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 13:21:29.981 NotebookApp] Forbidden
[W 13:21:29.982 NotebookApp] 403 GET /jupyter/api/sessions?_=1516645155611 (127.0.0.1) 2.00ms referer=http://localhost:8888/jupyter/tree/dask
[W 13:21:29.983 NotebookApp] Forbidden
[W 13:21:29.984 NotebookApp] 403 GET /jupyter/api/terminals?_=1516645155612 (127.0.0.1) 1.33ms referer=http://localhost:8888/jupyter/tree/dask
[I 13:21:32.678 NotebookApp] 302 GET /jupyter/ (127.0.0.1) 0.75ms
[I 13:21:32.691 NotebookApp] 302 GET /jupyter/tree? (127.0.0.1) 1.26ms
[I 13:21:35.938 NotebookApp] 302 POST /jupyter/login?next=%2Fjupyter%2Ftree%3F (127.0.0.1) 1.69ms
[W 13:21:47.203 NotebookApp] Notebook dask/dask_dataframe_csv_example.ipynb is not trusted
[I 13:21:48.558 NotebookApp] Kernel started: 00dc7556-393d-4aa9-8303-4e34767a602c
[W 13:21:48.639 NotebookApp] 404 GET /jupyter/nbextensions/dragdrop/main.js?v=20180122132127 (127.0.0.1) 14.70ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:21:48.805 NotebookApp] 404 GET /jupyter/nbextensions/search-replace/main.js?v=20180122132127 (127.0.0.1) 2.96ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:21:48.980 NotebookApp] 404 GET /nbextensions/snippets/snippets.json (127.0.0.1) 1.76ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[W 13:21:49.040 NotebookApp] 404 GET /jupyter/nbextensions/vim_binding/vim_binding.js?v=20180122132127 (127.0.0.1) 2.30ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
[I 13:21:49.550 NotebookApp] Adapting to protocol v5.1 for kernel 00dc7556-393d-4aa9-8303-4e34767a602c
[W 13:21:49.731 NotebookApp] 404 GET /jupyter/nbextensions/widgets/notebook/js/extension.js?v=20180122132127 (127.0.0.1) 1.41ms referer=http://localhost:8888/jupyter/notebooks/dask/dask_dataframe_csv_example.ipynb
```

I notice the difference between the 2 outputs is that version 5.3.1 makes 2 additional calls or requests to session/api/, one for sessions and one for terminals.  Not sure if that means anything or if it is of any significance.

**Comment by takluyver** [comment]
If you leave the extensions installed but disable them, does the problem go away? If so, try enabling them individually to narrow down where the problem is.

**Comment by pybokeh** [comment]
@takluyver Yes, I left them installed, just not enabled, and the notebooks functioned ok.  It took a while since I had several extensions enabled, and to be safe, I cleared my browser cache before enabling a new extension, but I narrowed down the culprit to be the snippets extension.  Should someone else duplicate before closing this issue to confirm?  Then, I should also open an issue at the notebook extensions repo.
